### PR TITLE
Extract Ledger API error definitions in //ledger/ledger-api-errors [DPP-1269]

### PR DESCRIPTION
**Pull request by tudor-da**
This PR:
* Extracts the Ledger API error code definitions to `//ledger/ledger-api-errors` (this new target depends on the LF goodies still)
* Keeps the common error codes that are already in use in client-related libraries in `//ledger/error` but removes Daml-LF dependencies (solves the problem of using `ledger-api-akka` without LF baggage)
* Adds TODO in code (with created GH issue) related to fully decoupling `//ledger/ledger-api-errors` from LF baggage (GH issue: https://github.com/digital-asset/daml/issues/15453)

Error codes changes:
* A new error codes group is introduced (`CommonErrors`) that are to be used in Ledger API client applications as well. These do not depend on LF. 4 generic errors are moved under that group. 
* `SERVICE_INTERNAL_ERROR` error code is created in `CommonErrors` as a fix for the usage of `LEDGER_API_INTERNAL_ERROR` in ledger api client

changelog_begin
`SERVICE_INTERNAL_ERROR` error code is created in CommonErrors and replaces the usage of `LEDGER_API_INTERNAL_ERROR` in ledger-api-client (as a fix).
changelog_end

### CODE REVIEW

File: docs/resources/generated-error-pages/error-codes-inventory.rst.inc [hunk 1]
```diff
@@ -359,54 +445,6 @@ PARTICIPANT_BACKPRESSURE
...
-
-    **Category**: TransientServerFailure
-
-    **Conveyance**: This error is logged with log-level INFO on the server side and exposed on the API with grpc-status UNAVAILABLE including a detailed error message.
-
-    **Resolution**: Retry re-submitting the request. If the error persists, contact the participant operator.
-
-
-
-
```
Comment (tudor-da) [inline docs/resources/generated-error-pages/error-codes-inventory.rst.inc:409]: Extracted in `CommonErrors`

File: docs/resources/generated-error-pages/error-codes-inventory.rst.inc [hunk 2]
```diff
@@ -228,13 +228,99 @@ RECORD_TIME_OUT_OF_RANGE
...
+SERVICE_INTERNAL_ERROR
+---------------------------------------------------------------------------------------------------------------------------------------
+    
+    **Explanation**: This error occurs if one of the services encountered an unexpected exception.
+
+    **Category**: SystemInternalAssumptionViolated
+
+    **Conveyance**: This error is logged with log-level ERROR on the server side. It is exposed on the API with grpc-status INTERNAL without any details for security reasons.
+
+    **Resolution**: Contact support.
```
Comment (tudor-da) [inline docs/resources/generated-error-pages/error-codes-inventory.rst.inc:281]: New INTERNAL error, which is more generic than the `LEDGER_API_INTERNAL_ERRO`

File: ledger/error/src/test/suite/scala/com/daml/error/utils/ErrorDetailsSpec.scala [hunk 3]
```diff
@@ -4,7 +4,7 @@
 package com.daml.error.utils
 
 import com.daml.error.ErrorCategory.BackgroundProcessDegradationWarning
-import com.daml.error.definitions.LedgerApiErrors
+import com.daml.error.definitions.CommonErrors
```
Comment (tudor-da) [inline ledger/error/src/test/suite/scala/com/daml/error/utils/ErrorDetailsSpec.scala:7]: Changed error examples used in this test in order to not depend on `//ledger/ledger-api-errors`

File: ledger/participant-integration-api/src/test/suite/scala/platform/apiserver/execution/LedgerTimeAwareCommandExecutorSpec.scala [hunk 4]
```diff
@@ -3,8 +3,7 @@
 
 package com.daml.platform.apiserver.execution
 
-import com.daml.error.definitions.ErrorCause
-import com.daml.error.definitions.ErrorCause.LedgerTime
+import com.daml.platform.apiserver.services.ErrorCause.LedgerTime
```
Comment (mziolekda) [inline ledger/participant-integration-api/src/test/suite/scala/platform/apiserver/execution/LedgerTimeAwareCommandExecutorSpec.scala:6]: Shouldn't this one be moved to the alphabetically justified slot too?

File: daml-lf/validation/BUILD.bazel [hunk 5]
```diff
@@ -22,7 +22,12 @@ da_scala_library(
     visibility = [
         "//compiler/scenario-service:__subpackages__",
         "//daml-lf:__subpackages__",
-        "//ledger/error:__subpackages__",
+        # TODO https://github.com/digital-asset/daml/issues/15453
+        #      Extract the error types in separate package
+        #      in order to decouple the error definitions and participant-integration-api
+        #      from unnecessary daml-lf/validation dependencies
+        "//ledger/ledger-api-errors:__subpackages__",
+        "//ledger/participant-integration-api:__subpackages__",
```
Comment (remyhaemmerle-da) [inline daml-lf/validation/BUILD.bazel:30]: Would it be fine to move the errors in the `daml-lf/language` module ? 
This module is mostly concern about the data structure to represent LF (+ some relatively small utility code).

File: ledger/participant-integration-api/src/test/suite/scala/platform/apiserver/execution/LedgerTimeAwareCommandExecutorSpec.scala (hunk 4 above)
Comment (tudor-da) [reply ledger/participant-integration-api/src/test/suite/scala/platform/apiserver/execution/LedgerTimeAwareCommandExecutorSpec.scala:6]: No idea 🤷‍♂️ I can check all files, but it's hard to enforce these rules unless an code formatter does it for you

File: ledger/error/BUILD.bazel [hunk 6]
```diff
@@ -17,16 +17,9 @@ da_scala_library(
...
     deps = [
-        "//daml-lf/archive:daml_lf_archive_reader",
-        "//daml-lf/data",
-        "//daml-lf/engine",
-        "//daml-lf/language",
-        "//daml-lf/transaction",
-        "//daml-lf/validation",
         "//ledger-api/grpc-definitions:ledger_api_proto_scala",
         "//ledger/ledger-grpc",
         "//ledger/participant-integration-api:participant-integration-api-proto_scala",
```
Comment (mziolekda) [inline ledger/error/BUILD.bazel:22]: Isn't that one still problematic?

File: docs/resources/generated-error-pages/error-codes-inventory.rst.inc (hunk 2 above)
Comment (mziolekda) [reply docs/resources/generated-error-pages/error-codes-inventory.rst.inc:281]: This is changing the behavior of the places in code that use `LEDGER_API_INTERNAL_ERROR` to `SERVICE_INTERNAL_ERROR`. As far as I can tell, this affects ledger api client code only, but I may be wrong about that. As a minimum, this needs to be documented in the release notes.

Comment (tudor-da) [reply docs/resources/generated-error-pages/error-codes-inventory.rst.inc:281]: Not all of them change to `SERVICE_INTERNAL_ERROR`, only the ones in the Ledger API client code (which I think is fine imho - error codes are not forwarded over the wire anyway for INTERNALs).

File: ledger/error/BUILD.bazel (hunk 6 above)
Comment (tudor-da) [reply ledger/error/BUILD.bazel]: Not really, it just pulls in some proto defs from the Index. It also seems that it the target compiles without it. Removed it

File: docs/resources/generated-error-pages/error-codes-inventory.rst.inc (hunk 2 above)
Comment (tudor-da) [reply docs/resources/generated-error-pages/error-codes-inventory.rst.inc:281]: I know that it's deprecated, but added it to the changelog. See ⬆️

**Review by mziolekda — APPROVED** [review]

**Pull request merged**